clash-protocols/src/Protocols/Df.hs

```diff
@@ -224,6 +220,14 @@ toData :: Bool -> a -> Data a
 toData False _ = NoData
 toData True a = Data a
 
+{- |
+  If the t'Data' is v'NoData', it returns the given value; otherwise,
+  it returns the value contained in the v'Data'.
+-}
+dataDefault :: a -> Data a -> a
+dataDefault a NoData = a
+dataDefault _ (Data a) = a
```

Comment:
`fromData` is arguably a more idiomatic name for this (c.f. `fromMaybe`).

Reply:
Agreed but that function exists already as:

```
-- | Extract value from `Data`, Bottom on `NoData`.
fromData :: (HasCallStack, C.NFDataX a) => Data a -> a
fromData NoData = C.deepErrorX "fromData: NoData"
fromData (Data a) = a
```